### DISCUSSION TITLE
Running this Client on MacOS

### DISCUSSION THREAD

**Discussion by strau0106** [abraunegg/onedrive#1905] [Q&A]
Hi Guys!
I am wondering if I can run this OneDrive client also on MacOS. I am currently Linux/MacOS dualboot on my Macbook Air8,2. 
(t2linux.org). 
My main concerns with the Microsoft client are that
- I can't replicate the docs/Sharepoint-Shared-libraries.md functionality
- It sucks extremely high amounts of preformance and leaks memory. 

That's why I'd rather not use the docker version because it's probably as preformance intensive as the Microsoft client....

I've tried to build already on MacOS.... I haven't seen any build instructions however. A simple make in the directory doesn't work....

```
if [ -f .git/HEAD ] ; then \
		git describe --tags > version ; \
	else \
		echo v2.4.16 > version ; \
	fi
dmd  -w -g -O -J. -L-lcurl -L-lsqlite3  -L-ldl src/config.d src/itemdb.d src/log.d src/main.d src/monitor.d src/onedrive.d src/qxor.d src/selective.d src/sqlite.d src/sync.d src/upload.d src/util.d src/progress.d src/arsd/cgi.d -ofonedrive
src/monitor.d(12): Error: undefined identifier `IN_CLOSE_WRITE`
src/monitor.d(12): Error: undefined identifier `IN_CREATE`
src/monitor.d(12): Error: undefined identifier `IN_DELETE`
src/monitor.d(12): Error: undefined identifier `IN_MOVE`
src/monitor.d(12): Error: undefined identifier `IN_IGNORED`
src/monitor.d(12): Error: undefined identifier `IN_Q_OVERFLOW`
src/monitor.d(243): Error: undefined identifier `inotify_event`
make: *** [onedrive] Error 1
```

If you can help me I'd be very happy. 

(About unit files and auto running that shouldn't be a problem. I can write a LaunchAgent plist...)
(I have my OneDrive on an ExFat partition on my Harddrive)

Best regards and any help is appreciated : )
Strau

**Comment by abraunegg** [comment]
@strau0106 

> That's why I'd rather not use the docker version because it's probably as preformance intensive as the Microsoft client....

I would suggest try on a Docker container on MacOS first and see how the performance is before jumping to such conclusion ..

All the compiler errors are because the code is incompatible with MacOS due to the inotify elements. The only way to resolve those would be to investigate options or criple the client when running on MacOS ... which, if that was the case - running under Docker is a much better proposition to retain functionality.

**Reply by strau0106** [reply]
Okay I'll try the docker later today. Also problem is storage on disk because I my linux volume already uses half...
But the inotify part makes sense....
Thanks alot for this amazing client!
Strau

**Reply by abraunegg** [reply]
> Okay I'll try the docker later today. Also problem is storage on disk because I my linux volume already uses half...

My suggestion then is:

* In your Linux volume use https://github.com/jstaf/onedriver so that you consume the 'on-demand' feature set
* Run https://github.com/jstaf/onedriver in your own Linux Docker container on the Mac so that you also consume the 'on-demand' feature set

If you actually need all your files, then my suggestion is somehow mount your 'OneDrive' data locally in a filesystem that both MacOS and Linux has access to, thus you only need 1 copy of your files locally.

**Reply by strau0106** [reply]
I already have it on a ExFat volume...

But i think ill just leave it as is... im on linux more than on macos
